vnext/Shared/OInstance.cpp

```diff
@@ -385,6 +389,24 @@ InstanceImpl::InstanceImpl(
           m_devSettings->jsiRuntimeHolder =
               std::make_shared<Microsoft::JSI::ChakraRuntimeHolder>(m_devSettings, m_jsThread, nullptr, nullptr);
           break;
+        case JSIEngineOverride::V8Napi: {
+#if defined(USE_V8)
+          std::unique_ptr<facebook::jsi::PreparedScriptStore> preparedScriptStore = nullptr;
+
+          char tempPath[MAX_PATH];
+          if (GetTempPathA(MAX_PATH, tempPath)) {
+            preparedScriptStore = std::make_unique<facebook::react::BasePreparedScriptStoreImpl>(tempPath);
+          }
+
+          m_devSettings->jsiRuntimeHolder = make_shared<NapiJsiV8RuntimeHolder>(
```

Comment:
if the above fails, you still run this and preparedScriptStore is null

Reply:
@tudorms should we explicitly fail if `preparedScriptStore` is not properly initialized?

(Note, this is the current exact behavior we have in plain `V8`).

Reply:
As a guiding principle, we shouldn't use "it's broken already in this other case", esp. when adding the same code / touching the same area. There are lots of PRs, and nobody can scale to look at every single one, so if someone spots a problem in a subsequent PR, we should fix it instead of deferring and making the problem worse in the meantime.